staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -1572,3 +1567,24 @@ func buildOpenAPIModelsForApply(staticOpenAPISpec *spec.Swagger, crd *apiextensi
 	}
 	return models, nil
 }
+
+func findCRD(crdName string, crds []*apiextensionsv1.CustomResourceDefinition) (*apiextensionsv1.CustomResourceDefinition, string, error) {
+	var crdClusterName string
+	var crd *apiextensionsv1.CustomResourceDefinition
+
+	for _, aCRD := range crds {
+		if aCRD.Name != crdName {
+			continue
+		}
+		if crd == nil {
+			crd = aCRD
+			crdClusterName = aCRD.GetClusterName()
+		} else {
+			if !equality.Semantic.DeepEqual(crd.Spec, aCRD.Spec) {
+				return nil, "", apierrors.NewInternalError(fmt.Errorf("error resolving resource: cannot watch across logical clusters for a resource type with several distinct schemas"))
```

Comment:
this is a little odd. Errors shouldn't be created by generic helpers. Would move this back into the call site. Same for the `aCRD.GetClusterName()`. That's a oneline for outside. What is left is nicely contained helper.

Reply:
`func findCRD(crdName string, crds []*apiextensionsv1.CustomResourceDefinition) (*apiextensionsv1.CustomResourceDefinition, bool)` would be the canonical signature.